apple/swift-docc-plugin
# Add --include-extended-types flag

Bug/issue #, if applicable: apple/swift-docc#210

## Summary

The `--include-extended-types` flag is transformed to the `--emit-extension-block-symbols` flag of the `swift package dump-symbol-graph` command. It is available starting from Swift 5.8.

## Dependencies

All dependencies have been merged. For an overview see https://github.com/apple/swift-docc/issues/210#issuecomment-1312692921.

## Testing

1. Use a toolchain generated from `main` or `release/5.8` branches after December 21st
2. Use the base branch of this PR when adding the `swift-docc-plugin` to your `Package.swift`:
```swift
dependencies: [
    .package(url: "https://github.com/themomax/swift-docc-plugin", branch: "add-extended-types-flag")
]
```
3. Add the `--include-extended-types` flag when running your command

Verifying the basic functionality can be done with [this](https://github.com/theMomax/swift-docc-sample-nested-types) sample repository. Running `swift package --disable-sandbox preview-documentation --include-extended-types --target NestedTypes` should produce the result shown below:

<img width="1210" alt="image" src="https://user-images.githubusercontent.com/21169289/201511142-a3e7c419-07eb-4fb3-b3c4-7ac92901192a.png">

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

Swift 5.7.2: `.bin/test` succeeded
Swift 5.8/main nightly: `.bin/test` succeeded after removing the `--parallel` flag from `swift test` invocations in the script (Running `swift test --parallel` on any package fails for me with an error similar to the one described [here](https://github.com/apple/swift-package-manager/issues/4562). However, this is an issue with the nightly toolchains I've been getting for a while now.)
